### PR TITLE
refactor(core): remove unused `CountersOwner::refund_gas`

### PR DESCRIPTION
**Pull request by SabaunT**
`SystemTerminationReason` error variants were completely unused, but removing it completely seems a hasty decision.

### CODE REVIEW

File: core-processor/src/ext.rs [hunk 1]
```diff
@@ -772,12 +763,12 @@ impl EnvExt for Ext {
     fn unreserve_gas(&mut self, id: ReservationId) -> Result<u64, Self::Error> {
         let amount = self.context.gas_reserver.unreserve(id)?;
 
-        // this statement is like in `Self::refund_gas()` but it won't affect "burned" counter
-        // because we don't actually refund we just rise "left" counter during unreservation
+        // This statement is like an op that increases `"eft" counter, but do not affect "burned" counter,
```
Comment (shamilsan) [inline core-processor/src/ext.rs:766]: `left`?
```suggestion
        // This statement is like an op that increases "left" counter but does not affect "burned" counter,
```

**Review by shamilsan — APPROVED** [review]

File: core-processor/src/ext.rs (hunk 1 above)
Comment (SabaunT) [reply core-processor/src/ext.rs]: fixed

**Review by grishasobol — APPROVED** [review]

**Review by grishasobol — CHANGES_REQUESTED** [review]
Please set label `A0-pleasereview`

**Review by grishasobol — CHANGES_REQUESTED** [review]
Conventional Commits

**Review by grishasobol — APPROVED** [review]

**Pull request merged**